### PR TITLE
Use nicer syntax in two places in CI

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
I've never seen a service watchdog get triggered before in CI, did you see this locally?

**Comment by keszybz** [comment]
> I've never seen a service watchdog get triggered before in CI, did you see this locally?

Yes. But there are also examples in CI: https://github.com/systemd/systemd/pull/35451 TEST-04-JOURNAL in `mkosi / ci (fedora, rawhide, address,undefined, 1, -Og, yes)`.

**Review by bluca — APPROVED** [review]

**Comment by keszybz** [comment]
Once again with different syntax. The previous attempt didn't actually pass the configuration option to the right place.

File: .github/workflows/mkosi.yml [hunk 1]
```diff
@@ -186,21 +187,25 @@ jobs:
...
+              -Dlibcryptsetup=enabled
+              -Dlibcurl=enabled
+              -Drepart=enabled
+              -Dfirstboot=true
+              -Dsysusers=true
+              -Dtmpfiles=true
+              -Dhwdb=true
               -Dvmspawn=enabled
+          )
+          [ -n ${{ matrix.sanitizers }} ] && OPTIONS+=( -Dservice-watchdog= )
```
Comment (yuwata) [inline .github/workflows/mkosi.yml]: Is this for #35335?
I do not think it is killed by the watchdog, but killed by TimeoutStopSec=, which was extended to 2min by b75befc3c98bcfa57ab801a24234ecab1c0b34a9.

Note, fedora has /usr/lib/systemd/system/service.d/10-timeout-abort.conf , so with the stop timeout, SIGABRT is sent to the service.

So, I think it is better to disable the drop-in 10-timeout-abort.conf when running with the sanitizers.

Hopefully, any failures in stopping services will be caught by sanitizers. So, not necessary to dump core.

Also, the drop-in config is Fedora specific, I guess that's why we do not see similar issue on CentOS CI with sanitizers.

**Review by yuwata — CHANGES_REQUESTED** [review]

File: .github/workflows/mkosi.yml (hunk 1 above)
Comment (yuwata) [reply .github/workflows/mkosi.yml]: -> #35460.

Comment (keszybz) [reply .github/workflows/mkosi.yml]: Yeah, that PR makes more sense.

The good news is that my syntax change is interpreted correctly:
```
   User defined options
    buildtype                                : debugoptimized
    blkid                                    : enabled
    firstboot                                : true
    hwdb                                     : true
    integration-tests                        : true
    libcryptsetup                            : enabled
    libcurl                                  : enabled
    openssl                                  : enabled
    remote                                   : enabled
    repart                                   : enabled
    service-watchdog                         : 
    sysusers                                 : true
    tmpfiles                                 : true
    tpm2                                     : enabled
    vmspawn                                  : enabled
```

I'll update with just the syntax change, to make future changes to options easier.

**Review by yuwata — APPROVED** [review]

**Pull request merged**